kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -0,0 +1,25 @@
+const Kolibri = require('kolibri');
+
+/**
+ * Action to fetch a particular content node from the API.
+ * @param {Function} dispatch - The dispatch method of the store object.
+ * @param {String} id - The id of the model to be fetched.
+ */
+const fetchFullContent = ({ dispatch }, id) => {
+  // Get the model from the contentmetadata resource.
+  const contentModel = Kolibri.resources.getResource('contentmetadata').getModel(id);
+  // Check to see if it is already synced from the server.
+  if (contentModel.synced) {
```

Comment:
I feel like we could make a nice idiom here along the lines of `contentModel.fetch_if_needed().then...`, to avoid the duplication, with promises being checked internally to determine whether to resolve immediately vs do the fetch and wait. Or perhaps it could be `contentModel.fetch({allow_caching: true}).then...`?